File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        m = 0
        while left < right:
            m = max((min(height[left], height[right]) * (right-left)),m)
            if height[left] <= height[right]:
                left += 1
            elif height[left] >= height[right]:
                right -=1
        return m

File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int pt1 = 0, pt2 = height.size() - 1;
        int maxArea = 0,area,length,width;
        while(pt2 > pt1){
            width = pt2 - pt1;
            area = width * min(height[pt1], height[pt2]);
            maxArea = max(maxArea, area);
            if(height[pt1] < height[pt2]){
                pt1++;
            }
          else{
                pt2--;
            }
        }
        return maxArea;
    }
};